dbal/dsn.go

```diff
@@ -15,11 +15,6 @@ const (
 // - shared but unique in the same process
 // see: https://sqlite.org/inmemorydb.html
 func IsMemorySQLite(dsn string) bool {
-	r := regexp.MustCompile(`sqlite://file::?\w+:\?_fk=true&*(cache=shared)?(mode=memory)?`)
-
-	if r.MatchString(dsn) {
-		return true
-	}
-
-	return false
+	r := regexp.MustCompile(`^(?:sqlite://((file:((:memory)|(\w+)):)|(:memory:))(\?.*))$|(?:^(memory|:memory:)$)`)
+	return r.MatchString(dsn)
```

Comment:
OK sorry to bother again, but I still see problems with this.
Why not solely rely on the rules presented in https://sqlite.org/inmemorydb.html ?
```suggestion
	if dsn == "memory" {
		return true
	}

	u, err := url.Parse(dsn)
	if err != nil {
		return false
	}

	return u.Scheme == "sqlite" && (u.Host == "file::memory:" || u.Host == ":memory:" || (u.Query().Get("mode") == "memory" && strings.HasPrefix(u.Host, "file:")))
```

The current pattern does not accept e.g. `sqlite://file:memdb1?mode=memory&cache=shared` (literally taken from the sqlite docs).